app/components/EmojiPicker.tsx

```diff
@@ -0,0 +1,229 @@
+import { useTranslation } from "react-i18next";
+import { usePopoverState, PopoverDisclosure } from "reakit/Popover";
+import styled, { useTheme } from "styled-components";
+import { depths, s } from "@shared/styles";
+import { hexToRgb } from "@shared/utils/color";
+import Button from "~/components/Button";
+import Popover from "~/components/Popover";
+import usePickerTheme from "~/hooks/usePickerTheme";
+import useUserLocale from "~/hooks/useUserLocale";
+import { hover } from "~/styles";
+import Emoji from "./Emoji";
+import NudeButton from "./NudeButton";
+
+type Props = {
+  value?: string | null;
+  onChange: (emoji: string | null) => Promise<void>;
+  onClickOutside: () => void;
+};
+
+const locales = {
+  cs_CZ: "cs",
+  de_DE: "de",
+  en_US: "en",
+  es_ES: "es",
+  fa_IR: "fa",
+  fr_FR: "fr",
+  it_IT: "it",
+  ja_JP: "ja",
+  ko_KR: "kr",
+  nl_NL: "nl",
+  pt_BR: "pt",
+  pt_PT: "pt",
+  pl_PL: "pl",
+  tr_TR: "tr",
+  vi_VN: "vi",
+  zh_CN: "zh",
+  zh_TW: "zh",
+};
+
+const DEFAULT_EMOJIS_PER_LINE = 9;
+
+function EmojiPicker({
+  value,
+  onChange,
+  onClickOutside,
+  ...pickerOptions
+}: Props) {
+  const { t } = useTranslation();
+  const pickerTheme = usePickerTheme();
+  const theme = useTheme();
+  const language = useUserLocale();
+  const locale = language && locales[language] ? locales[language] : "en";
+
+  const popover = usePopoverState({
+    placement: "bottom-start",
+    modal: true,
+  });
+
+  const [emojisPerLine, setEmojisPerLine] = React.useState(
+    DEFAULT_EMOJIS_PER_LINE
+  );
+
+  const pickerRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    if (popover.visible && pickerRef.current) {
+      // 28 is picker's observed width when perLine is set to 0
+      // and 36 is the default emojiButtonSize
+      // Ref: https://github.com/missive/emoji-mart#options--props
+      setEmojisPerLine(Math.floor((pickerRef.current.clientWidth - 28) / 36));
+    }
+  }, [popover.visible]);
+
+  const handleEmojiChange = React.useCallback(
+    async (emoji) => {
+      popover.hide();
+      const val = emoji ? emoji.native : null;
+      await onChange(val);
+    },
+    [popover, onChange]
+  );
+
+  const handleClick = React.useCallback(
+    (ev: React.MouseEvent) => {
+      ev.stopPropagation();
+      if (popover.visible) {
+        popover.hide();
+      } else {
+        popover.show();
+      }
+    },
+    [popover]
+  );
+
+  const handleClickOutside = React.useCallback(() => {
+    // It was observed that onClickOutside got triggered
+    // even when the picker wasn't open or opened at all.
+    // Hence, this guard here...
+    if (popover.visible) {
+      onClickOutside();
+    }
+  }, [popover.visible, onClickOutside]);
+
+  return (
+    <>
+      <PopoverDisclosure {...popover}>
+        {(props) => (
+          <EmojiButton size={32} {...props} onClick={handleClick}>
+            {value ? (
+              <Emoji size="24px" native={value} />
```

Comment:
It would be good to have some hover state on this to hint that it's clickable when an emoji already exists, even just making it larger on hover would help.

Reply:
How about showing a border upon hover?

Reply:
sure